src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -0,0 +1,660 @@
+                    foreach (var method in restClient.Methods)
+                    {
+                        var responseClassifierType = CreateResponseClassifierType(method);
+                        responseClassifierTypes.Add(responseClassifierType);
+                        RequestWriterHelpers.WriteRequestCreation(writer, method, "internal", fieldNames, responseClassifierType.Name, false);
+                    }
+
+                    foreach ((string name, StatusCodes[] statusCodes) in responseClassifierTypes.Distinct())
+                    {
+                        WriteResponseClassifier(writer, name, statusCodes);
+                    }
+                }
+            }
+        }
+
+        private static IReadOnlyDictionary<string, string> WriteClientFields(CodeWriter writer, LowLevelRestClient client, BuildContext context)
+        {
+            var fieldNames = new Dictionary<string, string>();
+
+            foreach (var scheme in context.CodeModel.Security.GetSchemesOrAnonymous())
+            {
+                switch (scheme)
+                {
+                    case AzureKeySecurityScheme azureKeySecurityScheme:
+                        writer.Line($"private const string {AuthorizationHeaderConstantName} = {azureKeySecurityScheme.HeaderName:L};");
+                        writer.Line($"private readonly {typeof(AzureKeyCredential)}? {KeyAuthFieldName};");
+                        break;
+                    case AADTokenSecurityScheme aadTokenSecurityScheme:
+                        writer.Append($"private static readonly string[] {ScopesConstantName} = ");
+                        writer.Append($"{{ ");
+                        foreach (var credentialScope in aadTokenSecurityScheme.Scopes)
+                        {
+                            writer.Append($"{credentialScope:L}, ");
+                        }
+                        writer.RemoveTrailingComma();
+                        writer.Line($"}};");
+                        writer.Line($"private readonly {typeof(TokenCredential)}? {TokenAuthFieldName};");
+                        break;
+                }
+            }
+
+            writer.Line();
+
+            writer.Line($"private readonly {typeof(HttpPipeline)} {PipelineField};");
+            writer.Line($"private readonly {typeof(ClientDiagnostics)} {ClientDiagnosticsField};");
+            foreach (Parameter clientParameter in client.Parameters)
+            {
+                fieldNames.Add(clientParameter.Name, "_" + clientParameter.Name);
+                writer.Line($"private readonly {clientParameter.Type} _{clientParameter.Name};");
+            }
+
+            writer.Line();
+
+            writer.WriteXmlDocumentationSummary($"The HTTP pipeline for sending and receiving REST requests and responses.");
+            writer.Line($"public virtual {typeof(HttpPipeline)} {PipelineProperty} {{ get => {PipelineField}; }}");
```

Comment:
Do we need both the field and the property?

Reply:
Probably not. Good catch.